### PR TITLE
feat(ingestion/dynamodb): implement pagination for list_tables

### PR DESCRIPTION
**Pull request by jinlintt**
Also addressed an issue in handling the `include_table_item` config where the same table names in two different regions could clash. The fix is to specify the table name in the format of `region.table_name` in the config and also does the matching this way in the code.

Lastly, use the `region.table_name` when applying the table pattern.

### CODE REVIEW

**Comment by jinlintt** [comment]
@asikowitz @mayurinehate could one of you take a look at this?

File: metadata-ingestion/src/datahub/ingestion/source/dynamodb/dynamodb.py [hunk 1]
```diff
@@ -175,39 +176,30 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
...
+                "dynamodb",
+                region_name=region,
+                aws_access_key_id=self.config.aws_access_key_id,
+                aws_secret_access_key=self.config.aws_secret_access_key.get_secret_value(),
+            )
+
+            for table_name in self._list_tables(dynamodb_client):
+                dataset_name = f"{region}.{table_name}"
+                if not self.config.table_pattern.allowed(dataset_name):
+                    logger.info(f"skipping table: {dataset_name}")
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/dynamodb/dynamodb.py:193]:
```suggestion
                    logger.debug(f"skipping table: {dataset_name}")
```

File: metadata-ingestion/src/datahub/ingestion/source/dynamodb/dynamodb.py [hunk 2]
```diff
@@ -175,39 +176,30 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
...
+                    self.report.report_dropped(dataset_name)
                     continue
+
+                logger.info(f"Processing table: {dataset_name}")
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/dynamodb/dynamodb.py:197]:
```suggestion
                logger.debug(f"Processing table: {dataset_name}")
```

File: metadata-ingestion/src/datahub/ingestion/source/dynamodb/dynamodb.py [hunk 3]
```diff
@@ -294,33 +302,38 @@ def construct_schema_from_dynamodb(
...
         response = dynamodb_client.batch_get_item(
             RequestItems={table_name: {"Keys": primary_key_list}}
-        ).get("Responses", None)
+        ).get("Responses")
         if response is None:
             logger.error(
                 f"failed to retrieve item from table {table_name} by the given key {primary_key_list}"
             )
             return
+        logger.info(
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/dynamodb/dynamodb.py:334]:
```suggestion
        logger.debug(
```

File: metadata-ingestion/src/datahub/ingestion/source/dynamodb/dynamodb.py [hunk 4]
```diff
@@ -175,39 +176,30 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
...
+            dynamodb_client = boto3.client(
+                "dynamodb",
+                region_name=region,
+                aws_access_key_id=self.config.aws_access_key_id,
+                aws_secret_access_key=self.config.aws_secret_access_key.get_secret_value(),
+            )
+
+            for table_name in self._list_tables(dynamodb_client):
+                dataset_name = f"{region}.{table_name}"
+                if not self.config.table_pattern.allowed(dataset_name):
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/dynamodb/dynamodb.py:192]: Would be helpful if table_pattern config description mentions that "pattern is matched against dynamodb table name formatted as `<region>.<table>`.

File: metadata-ingestion/src/datahub/ingestion/source/dynamodb/dynamodb.py [hunk 5]
```diff
@@ -294,33 +302,38 @@ def construct_schema_from_dynamodb(
     def include_table_item_to_schema(
         self,
         dynamodb_client: Any,
+        region: str,
         table_name: str,
         schema: Dict[str, SchemaDescription],
     ) -> None:
         """
-        It will look up in the config include_table_item dict to see if the current table name exists as key,
+        It will look up in the config include_table_item dict to see if "region.table_name" exists as key,
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/dynamodb/dynamodb.py:310]: This also needs to be updated in markdown document and example.

**Comment by jinlintt** [comment]
@mayurinehate thank you for reviewing. I have addressed your feedback. Please take another look when you get a chance.

**Review by mayurinehate — APPROVED** [review]
LGTM

**Comment by jinlintt** [comment]
Thank you @mayurinehate for reviewing. @asikowitz could you merge it?

**Review by asikowitz — APPROVED** [review]

**Comment by jinlintt** [comment]
@asikowitz @anshbansal can we merge this now that the CI all seem to pass.
The frontend docker image failure doesn't seem to be related.

**Pull request merged**